packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,513 @@
+              child: child,
+            ),
+          ),
+        ),
+      );
+      _gestureChildren.add(child);
+    }
+
+    final Widget box = new _SegmentedControlRenderWidget<T>(
+      children: _gestureChildren,
+      selectedIndex: _selectedIndex,
+      pressedIndex: _pressedIndex,
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget<T> extends MultiChildRenderObjectWidget {
+  _SegmentedControlRenderWidget({
+    Key key,
+    List<Widget> children = const <Widget>[],
+    @required this.selectedIndex,
+    @required this.pressedIndex,
+  }) : super(
+          key: key,
+          children: children,
+        );
+
+  final int selectedIndex;
+  final int pressedIndex;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl<T>(
+      textDirection: Directionality.of(context),
+      selectedIndex: selectedIndex,
+      pressedIndex: pressedIndex,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSegmentedControl<T> renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..selectedIndex = selectedIndex
+      ..pressedIndex = pressedIndex;
+  }
+}
+
+class _SegmentedControlContainerBoxParentData extends ContainerBoxParentData<RenderBox> {
+  RRect surroundingRect;
+}
+
+class _RenderSegmentedControl<T> extends RenderBox with
+    ContainerRenderObjectMixin<RenderBox, ContainerBoxParentData<RenderBox>>,
+    RenderBoxContainerDefaultsMixin<RenderBox, ContainerBoxParentData<RenderBox>>{
+  _RenderSegmentedControl({
+    List<RenderBox> children,
+    @required int selectedIndex,
+    @required int pressedIndex,
+    @required TextDirection textDirection,
+  })  : _textDirection = textDirection,
+        _selectedIndex = selectedIndex,
+        _pressedIndex = pressedIndex {
+    addAll(children);
+  }
+
+  int get selectedIndex => _selectedIndex;
+  int _selectedIndex;
+  set selectedIndex(int value) {
+    if (_selectedIndex == value) {
+      return;
+    }
+    _selectedIndex = value;
+    markNeedsPaint();
+  }
+
+  int get pressedIndex => _pressedIndex;
+  int _pressedIndex;
+  set pressedIndex(int value) {
+    if (_pressedIndex == value) {
+      return;
+    }
+    _pressedIndex = value;
+    markNeedsPaint();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  List<Paint> get fillPaint => _fillPaint;
+  final List<Paint> _fillPaint = <Paint>[];
+
+  @override
+  double computeMinIntrinsicWidth(double height) => constraints.maxWidth / childCount;
```

Comment:
`constraints` isn't available here, this method can be called before layout is started.

Also, I don't think this makes sense? The child width is `constraints.maxWidth / childCount`, not the width of the segmented control itself.

I think here you just want to iterate over all the children, find the child with the greatest minIntrinsicWidth at the given height, and then return childCount times that value.

Reply:
We want the segmented control to spam the entire width of the screen, minus some padding, regardless of the natural widths of the children. Multiplying the greatest child width by childCount might result in the segmented control not spamming the width of the screen